maths/binary_multiplication.py

```diff
@@ -24,6 +50,35 @@ def b_expo(a: int, b: int) -> int:
 
 
 def b_expo_mod(a: int, b: int, c: int) -> int:
+    """
+    Calculate the result of (a * b) % c using binary exponentiation and modular arithmetic.
+
+    Parameters:
+    a (int): The first number.
+    b (int): The second number.
+    c (int): The modulus.
+
+    Returns:
+    int: The result of (a * b) % c.
+
+    Examples:
+    >>> b_expo_mod(2, 3, 5)
+    1
+    >>> b_expo_mod(5, 0, 7)
+    0
+    >>> b_expo_mod(3, 4, 6)
+    0
+    >>> b_expo_mod(10, 5, 13)
+    8
```

Comment:
```suggestion
    >>> b_expo_mod(10, 5, 13)
    11
```
This doctest has been written incorrectly.
```
Failed example:
    b_expo_mod(10, 5, 13)
Expected:
    8
Got:
    11
```

Reply:
I fixed this. Thanks for your suggestion 🫶